integration-tests/src/setup/test_radio.rs

```diff
@@ -0,0 +1,753 @@
+        &indexer_address,
+        &graphcast_id,
+        &runtime_config.subgraphs.clone().unwrap_or(vec![
+            MOCK_SUBGRAPH_MAINNET.to_string(),
+            MOCK_SUBGRAPH_GOERLI.to_string(),
+        ]),
+        runtime_config.indexer_stake,
+        &runtime_config.poi,
+    )
+    .await;
+    setup_mock_env_vars(&mock_server_uri);
+
+    let private_key = env::var("PRIVATE_KEY").expect("No private key provided.");
+    let registry_subgraph =
+        env::var("REGISTRY_SUBGRAPH_ENDPOINT").expect("No registry subgraph endpoint provided.");
+    let network_subgraph =
+        env::var("NETWORK_SUBGRAPH_ENDPOINT").expect("No network subgraph endpoint provided.");
+    let graph_node_endpoint =
+        env::var("GRAPH_NODE_STATUS_ENDPOINT").expect("No Graph node status endpoint provided.");
+
+    if env::var("METRICS_PORT").is_ok() {
+        info!(
+            "Starting metrics server on port {}",
+            env::var("METRICS_PORT").unwrap()
+        );
+        tokio::spawn(handle_serve_metrics(
+            "0.0.0.0".to_string(),
+            env::var("METRICS_PORT")
+                .unwrap()
+                .parse::<u16>()
+                .expect("Failed to parse METRICS_PORT environment variable as u16"),
+        ));
+    }
+
+    let wallet = private_key.parse::<LocalWallet>().unwrap();
+    let mut rng = thread_rng();
+    let mut private_key = [0u8; 32];
+    rng.fill(&mut private_key[..]);
+
+    let private_key = SecretKey::from_slice(&private_key).expect("Error parsing secret key");
+    let private_key_hex = encode(private_key.secret_bytes());
+    env::set_var("PRIVATE_KEY", &private_key_hex);
+
+    let private_key = env::var("PRIVATE_KEY").unwrap();
+
+    // TODO: Add something random and unique here to avoid noise form other operators
+    let radio_name: &str = "test-poi-radio";
+
+    let my_address =
+        query_registry_indexer(registry_subgraph.clone(), graphcast_id_address(&wallet))
+            .await
+            .unwrap();
+    let my_stake = query_network_subgraph(network_subgraph.clone(), my_address.clone())
+        .await
+        .unwrap()
+        .indexer_stake();
+    info!(
+        "Initializing radio to act on behalf of indexer {:#?} with stake {}",
+        my_address.clone(),
+        my_stake
+    );
+
+    let graphcast_agent = GraphcastAgent::new(
+        private_key,
+        radio_name,
+        &registry_subgraph,
+        &network_subgraph,
+        &graph_node_endpoint,
+        vec![],
+        Some("testnet"),
+        runtime_config.subgraphs.clone().unwrap_or(vec![
+            MOCK_SUBGRAPH_MAINNET.to_string(),
+            MOCK_SUBGRAPH_GOERLI.to_string(),
+        ]),
+        None,
+        None,
+        Some(get_random_port()),
+        None,
+    )
+    .await
+    .unwrap();
+
+    _ = GRAPHCAST_AGENT.set(graphcast_agent);
+    _ = MESSAGES.set(Arc::new(SyncMutex::new(vec![])));
+
+    if runtime_config.is_setup_instance {
```

Comment:
What is the reason to specifically pass in a `empty_attestation_handler()`?

Reply:
we don't want the _setup_ instances to do comparisons, only the _check_ ones